File: companion/spss/python/b11.py
import pandas as pd
import numpy as np
from scipy import stats

veri = pd.read_csv("companion/spss/csv/b11.csv")
assert veri.shape == (395, 10)
assert not veri.isna().any().any()
assert veri.id.is_unique
assert veri.sex.isin([1, 2]).all()
assert veri.pass10.eq((veri.g3 >= 10).astype(int)).all()
grup_F = veri.loc[veri.sex == 1, "g3"]
grup_M = veri.loc[veri.sex == 2, "g3"]
fark = veri.g3-veri.g1
welch = stats.ttest_ind(grup_F, grup_M,
    equal_var=False, alternative="two-sided")
eslesmis = stats.ttest_rel(veri.g3, veri.g1,
                          alternative="two-sided")
print(welch); print(welch.confidence_interval(.95))
print(eslesmis); print(eslesmis.confidence_interval(.95))

def betimsel(degerler):
    return dict(n=len(degerler), ort=degerler.mean(),
        ss=degerler.std(ddof=1),
        sh=degerler.std(ddof=1)/np.sqrt(len(degerler)))

def test_ozeti(sonuc, ortalama_fark):
    aralik = sonuc.confidence_interval(.95)
    return dict(t=sonuc.statistic, df=sonuc.df,
        p_iki_yonlu=sonuc.pvalue, fark=ortalama_fark,
        alt=aralik.low, ust=aralik.high)

ozetler = {
    "F": betimsel(grup_F), "M": betimsel(grup_M),
    "Welch": test_ozeti(welch, grup_F.mean()-grup_M.mean()),
    "G3": betimsel(veri.g3), "G1": betimsel(veri.g1),
    "Eslesmis": test_ozeti(eslesmis, fark.mean()),
    "Fark": dict(betimsel(fark),
                 dz=fark.mean()/fark.std(ddof=1))}
for baslik, ozet in ozetler.items():
    print(baslik)
    for ad, deger in ozet.items():
        print(f"{ad}: {deger:.13f}")
print(int((veri.g3 == 0).sum()))

import matplotlib.pyplot as plt

sekil, eksenler = plt.subplots(1, 3, figsize=(12, 4))
eksenler[0].boxplot([grup_F, grup_M])
eksenler[0].set_xticks([1, 2], ["F", "M"])
eksenler[0].set_title("G3")
eksenler[1].boxplot(fark); eksenler[1].set_title("G3-G1")
eksenler[2].hist(fark, bins=15)
eksenler[2].set_title("G3-G1")
plt.tight_layout(); plt.show()

File: companion/spss/r/b11.R
veri <- read.csv("companion/spss/csv/b11.csv")
stopifnot(nrow(veri) == 395, ncol(veri) == 10,
  !anyNA(veri), !anyDuplicated(veri$id),
  all(veri$sex %in% c(1, 2)),
  all(veri$pass10 == as.integer(veri$g3 >= 10)))
grup_F <- veri$g3[veri$sex == 1]
grup_M <- veri$g3[veri$sex == 2]
fark <- veri$g3-veri$g1
welch <- t.test(grup_F, grup_M, var.equal=FALSE,
  alternative="two.sided", conf.level=.95)
eslesmis <- t.test(veri$g3, veri$g1, paired=TRUE,
  alternative="two.sided", conf.level=.95)
print(welch); print(eslesmis)

betimsel <- function(degerler) {
  c(n=length(degerler), ort=mean(degerler),
    ss=sd(degerler), sh=sd(degerler)/sqrt(length(degerler)))
}
test_ozeti <- function(sonuc, ortalama_fark) {
  c(t=unname(sonuc$statistic), df=unname(sonuc$parameter),
    p_iki_yonlu=sonuc$p.value, fark=ortalama_fark,
    alt=sonuc$conf.int[1], ust=sonuc$conf.int[2])
}
print(rbind(F=betimsel(grup_F), M=betimsel(grup_M)),
      digits=15)
print(test_ozeti(welch, mean(grup_F)-mean(grup_M)),
      digits=15)
print(rbind(G3=betimsel(veri$g3), G1=betimsel(veri$g1)),
      digits=15)
print(test_ozeti(eslesmis, mean(fark)), digits=15)
print(c(betimsel(fark), dz=mean(fark)/sd(fark)), digits=15)
print(sum(veri$g3 == 0))

par(mfrow=c(1, 3))
boxplot(g3 ~ sex, data=veri, names=c("F", "M"))
boxplot(fark, main="G3-G1")
hist(fark, main="G3-G1", xlab="Difference")
par(mfrow=c(1, 1))